test/FunctionalTests/AuthorizationTests.cs

```diff
@@ -87,5 +87,29 @@ public async Task CallAuthorizedServiceWithToken_Success()
 
             Assert.AreEqual(StatusCode.OK.ToTrailerString(), Fixture.TrailersContainer.Trailers[GrpcProtocolConstants.StatusTrailer].Single());
         }
+
+        [Test]
+        public async Task CallAuthorizedServiceWithInvalidToken_ReturnUnauthorized()
+        {
+            // Arrange
+
+            var requestMessage = new HelloRequest
+            {
+                Name = "World"
+            };
+
+            var ms = new MemoryStream();
+            MessageHelpers.WriteMessage(ms, requestMessage);
+
+            var httpRequest = new HttpRequestMessage(HttpMethod.Post, "Authorize.AuthorizedGreeter/SayHello");
+            httpRequest.Headers.Add("Authorization", $"Bearer SomeInvalidTokenHere");
+            httpRequest.Content = new GrpcStreamContent(ms);
+
+            // Act
+            var response = await Fixture.Client.SendAsync(httpRequest);
+
+            // Assert
+            Assert.AreEqual(HttpStatusCode.Unauthorized, response.StatusCode);
```

Comment:
a thought:  gRPC also has StatusCode.Unauthenticated (which is sent along with http status code 200), so the .NET Core built-in support for [Authorize]  doesn't quite behave as expected with gRPC. I'm not quite sure how to solve at this point, but we should discuss this in the sync meeting.
Would having a [GrpcAuthorize] attribute make sense?

Reply:
The attribute only specifies the policy name.

Is it important to return a gRPC status instead of 401? If so create an issue and we can discuss what we can do to improve it.